oriedita-ui/src/main/java/oriedita/editor/swing/dialog/PreferenceDialog.java

```diff
@@ -464,28 +473,116 @@ public void actionPerformed(ActionEvent e) {
+            JLabel nameLabel = getTextLabel(key);
+            listPanel.add(nameLabel, new GridConstraints(index, 1, 1, 1, GridConstraints.ANCHOR_NORTH, GridConstraints.FILL_HORIZONTAL, GridConstraints.SIZEPOLICY_WANT_GROW, 1, null, null, null, 0, false));
 
-            JButton keystrokeButton = getKeyStrokeButton(buttonService, frameProvider, rowIndex, key);
-            hotkeyPanel.add(keystrokeButton, new GridConstraints(rowIndex, 3, 1, 1, GridConstraints.ANCHOR_NORTH, GridConstraints.FILL_HORIZONTAL, GridConstraints.SIZEPOLICY_WANT_GROW, 1, null, null, null, 0, false));
+            JButton keystrokeButton = getKeyStrokeButton(buttonService, frameProvider, key);
+            listPanel.add(keystrokeButton, new GridConstraints(index, 3, 1, 1, GridConstraints.ANCHOR_NORTH, GridConstraints.FILL_HORIZONTAL, GridConstraints.SIZEPOLICY_WANT_GROW, 1, null, null, null, 0, false));
 
             //TODO: a restore default button for hotkeys specifically
         }
-        hotkeyPanel.add(spacer1, new GridConstraints(rowIndex, 0, 1, 3, GridConstraints.ANCHOR_CENTER, GridConstraints.FILL_VERTICAL, 1, GridConstraints.SIZEPOLICY_WANT_GROW, null, null, null, 0, false));
-        hotkeyPanel.add(spacer2, new GridConstraints(0, 2, ActionType.values().length, 1, GridConstraints.ANCHOR_CENTER, GridConstraints.FILL_HORIZONTAL, 1, GridConstraints.SIZEPOLICY_WANT_GROW, null, null, null, 0, false));
+        listPanel.add(spacer1, new GridConstraints(hotkeyCategoryMap.get(categoryHeader).size() - 1, 0, 1, 3, GridConstraints.ANCHOR_CENTER, GridConstraints.FILL_VERTICAL, 1, GridConstraints.SIZEPOLICY_WANT_GROW, null, null, null, 0, false));
+        listPanel.add(spacer2, new GridConstraints(0, 2, hotkeyCategoryMap.get(categoryHeader).size(), 1, GridConstraints.ANCHOR_CENTER, GridConstraints.FILL_HORIZONTAL, 1, GridConstraints.SIZEPOLICY_WANT_GROW, null, null, null, 0, false));
+
+    }
+
+    private void extractHeaders(List<String[]> allData) {
+        String[] headers = allData.get(0);
+        categoryHeaderList.addAll(Arrays.asList(headers));
+        for (String header : headers) {
+            hotkeyCategoryMap.put(header, new ArrayList<>());
+        }
+    }
+
+    private void extractData(List<String[]> allData) {
+        for (int i = 1; i < allData.size(); i++) {
+            String[] row = allData.get(i);
+            for (int j = 0; j < row.length; j++) {
+                if (!row[j].isEmpty()) {
+                    hotkeyCategoryMap.get(categoryHeaderList.get(j)).add(row[j]);
+                }
+            }
+        }
+    }
+
+    private void readCSV() {
+        try {
+            // Create an object of file reader class with CSV file as a parameter.
+            FileReader filereader = new FileReader("oriedita/src/main/resources/categories.csv");
```

Comment:
the file has to be loaded as a resource, otherwise it won't work in the exported .jar.
see https://stackoverflow.com/questions/15749192/how-do-i-load-a-file-from-resource-folder